### PR TITLE
Build from OpenMM repository

### PR DESCRIPTION
**Pull request by peastman**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)

The HIP platform is now in the main OpenMM repository.  I'm trying to get this package building from that repository rather than the separate StreamHPC one.

### CODE REVIEW

**Comment by peastman** [comment]
@conda-forge-admin, please rerender

**Comment by peastman** [comment]
@conda-forge-admin, please rerender

**Review by hmacdope — APPROVED** [review]
LGTM @peastman, as you say some things to iron out with a matching version, but should be good for now.

**Comment by peastman** [comment]
@conda-forge-admin, please rerender

File: .ci_support/linux_64_.yaml [hunk 1]
```diff
@@ -5,13 +5,13 @@ c_stdlib_version:
 cdt_name:
 - cos7
 channel_sources:
-- conda-forge
+- conda-forge/label/openmm_rc,conda-forge
 channel_targets:
-- conda-forge main
+- conda-forge openmm_rc
```
Comment (mikemhenry) [inline .ci_support/linux_64_.yaml:10]: We need to target the `rc` branch so we don't deploy this on main

Comment (mikemhenry) [reply .ci_support/linux_64_.yaml:10]: Just updated that, it was easier since the branch didn't exist

**Comment by mikemhenry** [comment]
Just waiting on these builds https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=1038423&view=results then it should be able to pull in 8.2.0beta

**Comment by peastman** [comment]
Do we really need separate `rc` and `main` branches?  In principle that makes it easier to create releases of one version and release candidates of a different version at the same time.  In practice, we never do that.  And it creates so many headaches switching back and forth between them.

**Comment by peastman** [comment]
Do we really only need a single build for this package?  I guess we don't need builds for different Python versions, because there are no Python components.  And HIP packages are only available for Linux, and only for x86.  I was just surprised to see it!

**Comment by mikemhenry** [comment]
> Do we really need separate rc and main branches? In principle that makes it easier to create releases of one version and release candidates of a different version at the same time. In practice, we never do that. And it creates so many headaches switching back and forth between them.

What it helps to prevent is if someone runs `micromamba install openmm-hip` it will pull down the "latest" version which could be `8.2.0beta` -- for this repo it matters less but had we just release `openmm 8.2.0beta` on `main` then right anyone that runs `micromamba install openmm` would pull down the `8.2.0beta` version instead of `8.1.1`

The trade off will come in support/issues raised IMHO, especially if there are any breaking changes.

**Comment by mikemhenry** [comment]
@conda-forge-admin, please rerender

**Comment by peastman** [comment]
I mean branches in the repo, not labels on the packages.  Definitely pre-release builds get posted to a different label.  But is there a good reason for building them off a separate `rc` branch?

**Comment by mikemhenry** [comment]
Other than that is how conda-forge says to do it: https://github.com/conda-forge/cfep/blob/main/cfep-05.md#specification I am not sure if there is a "good" reason.

I think it just keeps things tidy, just like how if you ever wanted to backport a fix to an older version its cleaner to do that on a branch than just on main. I am not sure if there is a technical reason or if can cause issues.

**Comment by peastman** [comment]
Every time we switch between the two branches, it produces dozens of merge conflicts that have to be resolved by hand.  Is there some better way of handling it?

**Comment by peastman** [comment]
Is this ready to merge?

**Comment by mikemhenry** [comment]
Yes this is ready to merge -- I will do it! 

> Every time we switch between the two branches, it produces dozens of merge conflicts that have to be resolved by hand. Is there some better way of handling it?

It was my fault for not _first_ rebasing the pr off of the `rc` branch instead of main. I'll admit that I made things a mess when I did that. In the future, I will do it on a separate PR so that things stay clean.

@conda-forge-admin 
I don't think the CFEP-05 makes it super clear, in RFC speak https://www.rfc-editor.org/rfc/rfc2119 is using a separate branch for an rc label a SHOULD, MUST, or MAY?

**Comment by peastman** [comment]
> It was my fault for not first rebasing the pr off of the rc branch instead of main.

I didn't mean that as a criticism of you.  The same thing happens every time I merge changes from `main` to `rc` or from `rc` to `main`.  If there's a better way of doing it that avoids merge conflicts, I'd love to know what it is.

**Comment by zklaus** [comment]
@peastman, I found in the past that merging with git on the command-line while employing different [merge strategies](https://git-scm.com/docs/merge-strategies) can be an effective tool. It's not without a learning curve and care is needed, but often a `git merge -s ours` works out. (Not to be confused with `git merge -X ours`, which, annoyingly, is something else entirely.)

**Pull request merged**